### PR TITLE
Add new target support to Error object and NativeErrors

### PR DESCRIPTION
**Pull request by kisbg**
EcmaScript 11 19.5.1.1 Error
EcmaScript 11 19.5.6.1.1 NativeError

JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

**Comment by LaszloLango** [comment]
Please rebase and update the commit message with the mention of the related section of ECMA standard.

**Review by LaszloLango — APPROVED** [review]
LGTM

**Comment by kisbg** [comment]
@LaszloLango  Thank you for the review!

File: jerry-core/ecma/builtin-objects/ecma-builtin-error.c [hunk 1]
```diff
@@ -62,7 +64,30 @@ ecma_value_t
...
+
+  if (proto_p == NULL)
+  {
+    return ECMA_VALUE_ERROR;
+  }
+
+  ecma_value_t result = ecma_builtin_error_dispatch_call (arguments_list_p, arguments_list_len);
+
+  if (ECMA_IS_VALUE_ERROR (result))
+  {
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-error.c:81]: Just to reduce code duplication:

```c
if (!ECMA_IS_VALUE_ERROR (result))
{
  ecma_object_t *object_p = ecma_get_object_from_value (result);
  ECMA_SET_NON_NULL_POINTER (object_p ->u2.prototype_cp, proto_p);
}

ecma_deref_object (proto_p);
return result;
```

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/ecma-builtin-error.c (hunk 1 above)
Comment (kisbg) [reply jerry-core/ecma/builtin-objects/ecma-builtin-error.c:81]: Good idea!

File: jerry-core/ecma/builtin-objects/ecma-builtin-evalerror.c [hunk 2]
```diff
@@ -64,7 +66,28 @@ ecma_value_t
...
+  ecma_value_t result = ecma_builtin_eval_error_dispatch_call (arguments_list_p, arguments_list_len);
+
+  if (!ECMA_IS_VALUE_ERROR (result))
+  {
+    ecma_object_t *object_p = ecma_get_object_from_value (result);
+    ECMA_SET_NON_NULL_POINTER (object_p->u2.prototype_cp, proto_p);
+  }
+
+  ecma_deref_object (proto_p);
+  return result;
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-evalerror.c:89]: New line before each return.

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

**Pull request merged**